### PR TITLE
support yuque && add `authHeaders` option

### PR DESCRIPTION
**Pull request by djyde**
This PR support a new provider [Yuque](https://yuque.com), which is a documentation platform from China.

According to Yuque's API authorization [definition](https://www.yuque.com/yuque/developer/api), every secured API call (such as the profile API) should pass access token in header `X-Auth-Token`, but I can't find an option that can pass the access token as a header value. So I add an `authHeaders` option in this PR.

### CODE REVIEW

File: types/providers.d.ts [hunk 1]
```diff
@@ -99,6 +100,7 @@ export type OAuthProviderType =
   | "WordPress"
   | "WorkOS"
   | "Yandex"
+  | "Yuque"
```
Comment (lluia) [inline types/providers.d.ts:103]: Can you add [type tests](https://github.com/nextauthjs/next-auth/blob/788d7dc41f8767a4659136475189d6d420295855/types/tests/providers.test.ts#L259) for **Yuque** as well? 😊

File: types/providers.d.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ export interface OAuthConfig<P extends Record<string, unknown> = Profile>
   requestTokenUrl?: string
   authorizationUrl: string
   profileUrl: string
+  authHeaders?(info: { accessToken: string }): Record<string, string>,
```
Comment (lluia) [inline types/providers.d.ts:36]: You get any tokens returned from the provider in the second param of the `profile` callback:
```ts
async profile(profile, tokens) {
  const data = await getDataFromApi(tokens.access_token)
  return {
    id: data.sub,
    name: data.userName,
    image: data.picture
  }
}
// ...
async getDataFromApi(token) {
  return fetch('path/to/api', {
    headers: {
      'X-Auth-Token': token
    }
  })
} 
```
Users can use that to make API calls to the provider. This is [documented in our options](https://next-auth.js.org/configuration/providers#options), although maybe we should highlight it better 🤔 as it's a common use case.

File: www/docs/providers/yuque.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  Providers.Yuque({
+    clientId: process.env.YUQUE_CLIENT_ID,
+    clientSecret: process.env.YUQUE_CLIENT_SECRET
+  })
+]
+...
+```
+
+:::warning
+Yuque doesn't expose user's email in API
```
Comment (lluia) [inline www/docs/providers/yuque.md:37]:
```suggestion
Yuque doesn't expose the user's email in their API
```

**Review by 0ubbe — CHANGES_REQUESTED** [review]
Thanks ✨ , few amendments before we can merge 😊

File: types/providers.d.ts (hunk 2 above)
Comment (djyde) [reply types/providers.d.ts:36]: This is used in https://github.com/djyde/next-auth/blob/69f1559ced9bc130238fd09c85c3a01578c9e6eb/src/server/lib/oauth/client.js#L222

**Pull request closed without merging**